whittle/models/gpt/extract.py

```diff
@@ -8,9 +9,24 @@
 from whittle.models.gpt.blocks.mlp import GptNeoxMLP, LLaMAMLP
 from whittle.modules.layernorm import LayerNorm
 from whittle.modules.rmsnorm import RMSNorm
+from litgpt import Config
+
+
+def extract_sub_network(model: GPT, sub_network_config: Config) -> GPT:
+    """
+    Extracts a sub-network from a given model based on the specified sub-network configuration.
+    Copies relevant layers, weights, and configurations from the full model into a sub-network model.
+
+    Args:
+        model: The original, full GPT model from which the sub-network is extracted.
+        sub_network_config: Configuration object for the sub-network, containing the necessary
+                                     architecture specifications such as embedding size, number of heads,
+                                     and number of layers.
 
+    Returns:
+        GPT: A new sub-network model instance, initialized with parameters extracted from the original model.
```

Comment:
```suggestion
        A new sub-network model instance, initialized with parameters extracted from the original model.
```

I would encourage to not include the type of the return or args in the docstring, if the type in the function signature changes you'll have to update two things which can often be overlooked by accident. Only adding the types in the function signature is therefore more maintainable :)